### PR TITLE
Minor fix to providers

### PR DESCRIPTION
**Pull request by fernandog**

### CODE REVIEW

File: medusa/providers/nzb/newznab.py [hunk 1]
```diff
@@ -359,6 +359,10 @@ def _match_indexer(self):
 
         return_mapping = {}
 
+        if not self.show:
+            # If we don't have show, can't get tvdbid
+            return {}
```
Comment (labrys) [inline medusa/providers/nzb/newznab.py]: this should return `return_mapping`

File: medusa/providers/nzb/newznab.py [hunk 2]
```diff
@@ -359,6 +359,10 @@ def _match_indexer(self):
...
+
         if not self.cap_tv_search or self.cap_tv_search == 'True':
             # We didn't get back a supportedParams, lets return, and continue with doing a search string search.
             return {}
```
Comment (labrys) [inline medusa/providers/nzb/newznab.py]: Same here

File: medusa/providers/torrent/json/rarbg.py [hunk 3]
```diff
@@ -101,7 +101,7 @@ def search(self, search_strings, age=0, ep_obj=None):
             else:
                 search_params['sort'] = self.sorting if self.sorting else 'seeders'
                 search_params['mode'] = 'search'
-                search_params['search_tvdb'] = self._get_tvdb_id()
+                search_params['search_tvdb'] = self._get_tvdb_id() if ep_obj else None
```
Comment (labrys) [inline medusa/providers/torrent/json/rarbg.py]: This should not be here.  Logic to test if show exists should be in `_get_tvdb_id` as it applies to all `_get_tvdb_id` calls.  And `_get_tvdb_id` should return falsey (`None` would probably be most appropriate) if an ID isn't available.

**Comment by labrys** [comment]
@pymedusa/developers anyone using non-tvdb mapped series that can test this?

**Comment by p0psicles** [comment]
Aah wait didn't meant to approve entire Pr.

**Comment by p0psicles** [comment]
Ill test it

**Comment by fernandog** [comment]
@p0psicles did you test this?

**Comment by fernandog** [comment]
@p0psicles any news on this?

**Comment by fernandog** [comment]
@medariox which test do you expect me to add?

If we don't have if show is None, the `self.show.indexerid` will fails

**Comment by p0psicles** [comment]
OK next on my list.

**Pull request merged**